Math/src/test/java/org/tribuo/math/neighbor/TestNeighborsBruteForce.java

```diff
@@ -0,0 +1,320 @@
+
+        assertNeighborDistances(indexDistancePairList, expectedDistance0, expectedDistance1, expectedDistance2);
+
+        //////////////////////////////////////////////
+        // Check the third point provided to the query
+        indexDistancePairList = indexDistancePairListArray[2];
+
+        // These tests use NUM_NEIGHBORS_K = 3 throughout. These are the expected neighboring points
+        double[] thirdExpectedPoint0 = {3.25,3.1};
+        double[] thirdExpectedPoint1 = {3.2,3.25};
+        double[] thirdExpectedPoint2 = {5.05,3.12};
+
+        assertNeighborPoints(data, indexDistancePairList, thirdExpectedPoint0, thirdExpectedPoint1, thirdExpectedPoint2);
+
+        // These tests use NUM_NEIGHBORS_K = 3 throughout. These are the expected neighboring distances
+        expectedDistance0 = 0.8805679985100522;
+        expectedDistance1 = 0.9769339793455846;
+        expectedDistance2 = 0.9823441352194252;
+
+        assertNeighborDistances(indexDistancePairList, expectedDistance0, expectedDistance1, expectedDistance2);
+    }
+
+    @Test
+    public void testSingleThreadQueryMany() {
+        SGDVector[] data = getTestDataVectorArray();
+        SGDVector[] queryData = getTestQueryVectorArray();
+        NeighborsBruteForce nbf = new NeighborsBruteForce(data, DistanceType.EUCLIDEAN, 1);
+        neighborsBrutForceQueryMany(nbf, data, queryData);
+    }
+
+    @Test
+    public void testMultiThreadQueryMany() {
+        SGDVector[] data = getTestDataVectorArray();
+        SGDVector[] queryData = getTestQueryVectorArray();
+        NeighborsBruteForce nbf = new NeighborsBruteForce(data, DistanceType.EUCLIDEAN, 2);
+        neighborsBrutForceQueryMany(nbf, data, queryData);
+    }
+
+    @Test
+    public void testNeighborsBrutForceQueryOne() {
+        SGDVector[] data = getTestDataVectorArray();
+        SGDVector vector = get2DPoint(5.21,5.28);
+        NeighborsBruteForce nbf = new NeighborsBruteForce(data);
+        List<Pair<Integer, Double>> indexDistancePairList = nbf.query(vector, NUM_NEIGHBORS_K);
+
+        // These tests use NUM_NEIGHBORS_K = 3 throughout. These are the expected neighboring points
+        double[] firstExpectedPoint0 = {5.23,5.02};
+        double[] firstExpectedPoint1 = {4.95,5.25};
+        double[] firstExpectedPoint2 = {5.01,5.03};
+
+        assertNeighborPoints(data, indexDistancePairList, firstExpectedPoint0, firstExpectedPoint1, firstExpectedPoint2);
+
+        // These tests use NUM_NEIGHBORS_K = 3 throughout. These are the expected neighboring distances
+        double expectedDistance0 = 0.2607680962081067;
+        double expectedDistance1 = 0.26172504656604784;
+        double expectedDistance2 = 0.32015621187164256;
+
+        assertNeighborDistances(indexDistancePairList, expectedDistance0, expectedDistance1, expectedDistance2);
+    }
+
+    private SGDVector[] getTestDataVectorArray() {
+        List<SGDVector> vectorList = new ArrayList<>();
+
```

Comment:
This is the same set of points used in Hdbscan tutorial. I didn't see an easy way to get points from a csv into `SGDVector`s

Reply:
Sure, we'll see if there is further need for this. Just wanted to make sure I hadn't missed it.